p2p/peer_pool.py

```diff
@@ -502,7 +506,13 @@ async def connect_to_node(self, node: NodeAPI) -> None:
                 async with self._connection_attempt_lock:
                     peer = await self.connect(node)
             except ALLOWED_PEER_CONNECTION_EXCEPTIONS:
-                return
+                if self._suppress_allowed_connection_exceptions:
+                    # These are all logged in self.connect(), so we simply return.
+                    return
+                raise
```

Comment:
*nitpick* style guide suggests doing the explicit branch here, like:
```suggestion
                else:
                    raise
```

Reply:
:+1: